File: factors.py
import sys


# Function to factorize a number n into two smaller numbers
def factorize(n):
    factors = []
    for i in range(2, n + 1):
        while n % i == 0:
            factors.append(i)
            n = n // i
    return factors


# Check if the command-line argument is provided
if len(sys.argv) != 2:
    print("Usage: factors <file>")
    sys.exit(1)

input_file = sys.argv[1]

try:
    with open(input_file, 'r') as file:
        for line in file:
            n = int(line.strip())
            if n <= 1:
                continue  # Skip numbers less than or equal to 1
            factors = factorize(n)
            if len(factors) >= 2:
                p, q = factors[0], factors[1]
                print(f"{n} = {p} * {q}")

except FileNotFoundError:
    print(f"File '{input_file}' not found.")
except ValueError:
    print("Invalid input in the file.")
except Exception as e:
    print(f"An error occurred: {e}")

File: factors.c
#include <stdio.h>
#include <stdlib.h>

// Function to factorize a number n into two smaller numbers
void factorize(int n) {
    int p, q;
    for (p = 2; p <= n / 2; ++p) {
        if (n % p == 0) {
            q = n / p;
            printf("%d = %d * %d\n", n, p, q);
            return;
        }
    }
    printf("%d is a prime number.\n", n);
}

int main(int argc, char *argv[]) {
    if (argc != 2) {
        printf("Usage: %s <file>\n", argv[0]);
        return 1;
    }

    char *input_file = argv[1];
    FILE *file = fopen(input_file, "r");

    if (file == NULL) {
        printf("File '%s' not found.\n", input_file);
        return 1;
    }

    int n;
    while (fscanf(file, "%d", &n) == 1) {
        if (n <= 1) {
            continue;  // Skip numbers less than or equal to 1
        }
        factorize(n);
    }

    fclose(file);
    return 0;
}
